src/components/form/form.tsx

```diff
@@ -234,4 +243,38 @@ export class Form {
             };
         });
     }
+
+    private getCustomErrorMessages(originalErrors: AjvError[]): AjvError[] {
+        if (!this.transformErrors) {
+            return originalErrors;
+        }
+
+        const errors: FormError[] = originalErrors.map((error: AjvError) => {
+            return {
+                name: error.name,
+                property: error.property,
+                message: error.message,
+                // For some reason 'schemaPath' is missing from the AjvError type definition:
+                // https://github.com/rjsf-team/react-jsonschema-form/issues/2140
+                // eslint-disable-next-line @typescript-eslint/dot-notation
+                schemaPath: error['schemaPath'],
+            };
+        });
+
+        // Use `.call({}, …)` here to bind `this` to an empty object to prevent
+        // the consumer submitted `transformErrors` from getting access to our
+        // component's internals. /Ads
+        return this.transformErrors
+            .call({}, errors)
```

Comment:
@jgroth I think we probably have more places where we call a consumer supplied callback as a method on `this`, and we need to use this pattern on every single one of them, or we have a security problem. I don't think we need to go find every instance right now, but we should at least keep this in the back of our heads and make sure to fix existing instances when we come across them, and also not let new instances in.

I've tested what happens if the callback is created using an arrow function, or if the consumer uses `bind(this)` before sending in the callback to limel-form, and that still works as expected—the `.call({}, …)` doesn't destroy the existing binding.

cc @BregenzerK FYI

Reply:
I don't think there are that many places actually :slightly_smiling_face: There's the `propFactory` that's also part of the form component and the `searcher` that's part of the picker. Besides those I don't think we have props that are functions. We might have some that are part of an inner structure of some prop, but I doubt we would bind it to the component in those cases :sweat_smile: 

Another way to get rid of the `this` binding is to simply assign the function to a constant/variable before invoking it, e.g.
```ts
const transformErrors = this.transformErrors;
transformErrors();
```

Reply:
Oh, that's a nifty solution. Nice!

Reply:
Sorry, I need some more context here. Is there anything we need to check and refactor in our add-ons?

Reply:
@BregenzerK Sorry, I mentioned you since you're a Lime Elements maintainer. I should have mentioned the group instead, to make the intention clear. I don't think you need to worry about this in add-ons, no.

Reply:
> I don't think there are that many places actually 🙂 There's the `propFactory` that's also part of the form component and the `searcher` that's part of the picker. Besides those I don't think we have props that are functions. We might have some that are part of an inner structure of some prop, but I doubt we would bind it to the component in those cases 😅

I agree we probably don't have many. We might have none actually, when I think about it. But we should at least keep in mind that this pattern could be dangerous, and that doing a `bind(this)` isn't necessary to open the vulnerability; it's enough to have a callback sent in as a property, and then calling it directly from `this` (like `this.consumerProvidedCallback()` for example).

> Another way to get rid of the `this` binding is to simply assign the function to a constant/variable before invoking it, e.g.
> 
> ```ts
> const transformErrors = this.transformErrors;
> transformErrors();
> ```

Nice 👍

Reply:
> > Another way to get rid of the `this` binding is to simply assign the function to a constant/variable before invoking it, e.g.
> > ```ts
> > const transformErrors = this.transformErrors;
> > transformErrors();
> > ```
> 
> Nice 👍

This one might run a greater risk of being "refactored" into the vulnerable version though… 🤔

It's pretty easy to write a test to check that the provided callback doesn't have access to the inner scope though, and as long as we do that, I like your suggestion 👍